app/controllers/posts_controller.rb

```diff
@@ -37,6 +38,32 @@ def index
 
   def homepage
     @latest_post = Post.order(sticky: :desc, created_at: :desc).first
+    @preview = params[:preview] == "1" && current_user.can_administrate_livestream?
+  end
+
+  def livestream_management
+  end
+
+  def update_test_link
+    new_value = params[:new_test_value]
+    test = ServerSetting.find_or_create_by(name: "TEST_wc2025_video_url")
+
+    if test.update!(value: new_value)
```

Comment:
This check is redundant in the sense that the `else` will never be entered: `update!` with the exclamation sign throws hard errors when something didn't work, so the code execution will abort and render the global "Something went wrong" error page.
Judging by your if-else-skeleton, you probably want the "soft" `update` method instead, which just returns `false` instead of raising the error

Reply:
```suggestion
    if test.update(value: new_value)
```

Reply:
Fixed